src/master.rs

```diff
@@ -18,13 +61,54 @@ where
         Driver::send_wakeup(self)
     }
 
-    fn write_frame(&mut self, pid: u8, data: &[u8]) -> Result<(), Driver::Error> {
-        self.send_header(pid)?;
-        self.write(data)
+    fn write_frame(&mut self, frame: &Frame) -> Result<(), Driver::Error> {
+        self.send_header(frame.get_pid())?;
+        self.write(frame.get_data_with_checksum())
     }
 
-    fn read_frame(&mut self, pid: u8, buf: &mut [u8]) -> Result<(), Driver::Error> {
+    fn read_frame(&mut self, pid: u8, data: &mut [u8]) -> Result<(), Driver::Error> {
```

Comment:
in my opinion it would be cleaner to return a `Frame` instead of a data buffer which has to be parsed by the caller

Reply:
Actually the data now returned is just the data without the checksum. But there is no reason not too return the whole frame actually.

Reply:
Done.